lib/js_es6/src/thrift.js

```diff
@@ -357,9 +358,14 @@ Thrift.TXHRTransport.prototype = {
               (function() {
                 var clientCallback = callback;
                 return function() {
-                  if (this.readyState == 4 && this.status == 200) {
-                    self.setRecvBuffer(this.responseText);
-                    clientCallback();
+                  if (this.readyState == 4 && !callbackWasCalled) {
+                    callbackWasCalled = true;
+                    if (this.status == 200) {
+                      self.setRecvBuffer(this.responseText);
+                      clientCallback(null);
+                    } else {
+                      clientCallback(new Error('Unexpected status: ' + this.status));
+                    }
```

Comment:
In my testing I saw `onerror` trigger before this callback was called with `this.readyState == 4` and also after. It depends on the nature of the error (no internet, server interrupt, no response, etc.), and even then I don't think we can guarantee behavior.